x/oracle/types/params.go

```diff
@@ -0,0 +1,131 @@
+package types
+
+import (
+	"fmt"
+	"github.com/btcsuite/btcd/btcec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"time"
+)
+
+var (
+	KeyOraclePublicKey          = []byte("OraclePublicKey")
+	KeyOraclePubKeyRemoteReport = []byte("OraclePubKeyRemoteReport")
+	KeyUniqueID                 = []byte("UniqueID")
+	KeyVoteParams               = []byte("VoteParams")
+	KeySlashParams              = []byte("SlashParams")
+)
+
+var _ paramtypes.ParamSet = (*Params)(nil)
+
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func DefaultParams() Params {
+	return Params{
+		OraclePublicKey:          nil,
+		OraclePubKeyRemoteReport: nil,
+		UniqueId:                 "",
+		VoteParams: VoteParams{
+			VotingPeriod: 30 * time.Second,
+			JailPeriod:   10 * time.Minute,
+			Quorum:       sdk.NewDecWithPrec(1, 3),
+		},
+		SlashParams: SlashParams{
+			SlashFractionDowntime: sdk.NewDecWithPrec(2, 1),
+			SlashFractionForgery:  sdk.NewDecWithPrec(1, 1),
+		},
+	}
+}
+
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyOraclePublicKey, &p.OraclePublicKey, validateOraclePublicKey),
+		paramtypes.NewParamSetPair(KeyOraclePubKeyRemoteReport, &p.OraclePubKeyRemoteReport, validateOraclePubKeyRemoteReport),
+		paramtypes.NewParamSetPair(KeyUniqueID, &p.UniqueId, validateUniqueID),
+		paramtypes.NewParamSetPair(KeyVoteParams, &p.VoteParams, validateVoteParams),
+		paramtypes.NewParamSetPair(KeySlashParams, &p.SlashParams, validateSlashParams),
+	}
+}
+
+func (p Params) Validate() error {
```

Comment:
Do you have any specific reason for excluding `validateOraclePubKeyRemoteReport` and `validateUniqueID`?

Reply:
The reason is that `validateOraclePubKeyRemoteReport` and `validateUniqueID` are values that cannot be validated by panacea.
I'm not sure what to validate when these two values are non-empty. :)